### PR TITLE
fix(ui5-date-picker): handle date selection via enter keyboard key

### PR DESCRIPTION
**Pull request by unazko**
- We don't need explicit handling on enter keyboard key
press after adaptations in the ui5-input control related
to change event handling.

Fixes: #4826

### CODE REVIEW

File: packages/main/src/DatePicker.js [hunk 1]
```diff
@@ -472,10 +471,6 @@ class DatePicker extends DateComponentBase {
 			return;
 		}
 
-		if (isEnter(event)) {
-			this._updateValueAndFireEvents(event.target.value, true, ["change", "value-changed"]);
```
Comment (ilhan007) [inline packages/main/src/DatePicker.js:476]: I see that this has been added on purpose by team Rila with this PR https://github.com/SAP/ui5-webcomponents/pull/4129 Could you please check with Team Rila about this?

**Review by a-z-ivanov — APPROVED** [review]

File: packages/main/src/DatePicker.js (hunk 1 above)
Comment (MapTo0) [reply packages/main/src/DatePicker.js:476]: We have added this due to some tests failings.
If you think you are OK after removing it - we are fine.

Angular's two way binding is dependent on `value-changed` and I would suggest you to recheck that scenario.

Comment (ilhan007) [reply packages/main/src/DatePicker.js:476]: @unazko as a follow up on the Martin's remark, you just need to ensure that on Enter, now that you remove is handling, the "change", and "value-changed" events are still fired. If fired we are fine with the Angular use case.

Comment (unazko) [reply packages/main/src/DatePicker.js:476]: I haven't tested this scenario with an Angular example, but we should be fine, as in the ui5-date-picker code <code>change</code> event and </code>value-changed</code> are always fired simultaneously via the <code>_updateValueAndFireEvents</code> function.

**Pull request merged**